consensus/state.go

```diff
@@ -354,6 +355,30 @@ func (cs *State) OnStart() error {
 		return err
 	}
 
+	// Double Signing Risk Reduction Logic
+	// triggered only when validator mode(existing privValidator) and DoubleSignCheckHeight set
+	if !(cs.privValidator == nil) && cs.config.DoubleSignCheckHeight > 0 && cs.Height > 0 {
+		valAddr, err := cs.privValidator.GetPubKey()
+		if err != nil {
+			return err
+		}
+
+		doubleSignCheckHeight := cs.config.DoubleSignCheckHeight
+		if doubleSignCheckHeight > cs.Height {
+			doubleSignCheckHeight = cs.Height
+		}
+		for i := int64(1); doubleSignCheckHeight > i; i++ {
+			lastCommit := cs.blockStore.LoadSeenCommit(cs.Height - i)
+			if lastCommit != nil {
+				for _, s := range lastCommit.Signatures {
+					if s.BlockIDFlag == types.BlockIDFlagCommit && bytes.Equal(s.ValidatorAddress, valAddr.Address()) {
+						cs.Logger.Error("Error checking double sign risk reduction logic", "err", ErrDoubleSignRiskReduction)
+						return ErrDoubleSignRiskReduction
```

Comment:
I can't find a place where we panic...

Reply:
when return error on there `func (cs *State) OnStart() error`, panic occurs in [`SwitchToConsensus`](https://github.com/tendermint/tendermint/blob/2ac5a559b4c88385e173b673d1f6250ff7c55263/consensus/reactor.go#L125-L127)

log example
```
I[2020-07-24|08:04:15.056] Started node                                 module=main nodeInfo="{ProtocolVersion:{P2P:9 Block:12 App:1} DefaultNodeID:fe0092406535cb943f5e15e3c6c33718089e1d3a ListenAddr:tcp://0.0.0.0:26656 Network:chain-hsa2Bm Version:0.33.6 Channels:40202122233038606100 Moniker:8CBB29C542D2D308 Other:{TxIndex:on RPCAddress:tcp://0.0.0.0:26657}}"
I[2020-07-24|08:04:15.367] Executed block                               module=state height=37 validTxs=0 invalidTxs=0
I[2020-07-24|08:04:15.367] Committed state                              module=state height=37 txs=0 appHash=0000000000000000
panic: Failed to start consensus state: double sign detected or restarted in DoubleSignCheckHeight

conS:
ConsensusState

conR:
ConsensusReactor

goroutine 45 [running]:
github.com/tendermint/tendermint/consensus.(*Reactor).SwitchToConsensus(0xc00025f880, 0xc, 0x1, 0xc0002a8440, 0x6, 0xc0002a8450, 0xc, 0x25, 0xc00258ebc0, 0x20, ...)
	github.com/tendermint/tendermint/consensus/reactor.go:127 +0x2cb
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0xc000b44700, 0xc0024af800)
	github.com/tendermint/tendermint/blockchain/v0/reactor.go:319 +0x11be
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
	github.com/tendermint/tendermint/blockchain/v0/reactor.go:110 +0x89
```